callouts/go/extproc/internal/server/callout_server.go

```diff
@@ -0,0 +1,229 @@
+		return
+	}
+	lis, err := net.Listen("tcp", s.Config.InsecureAddress)
+	if err != nil {
+		log.Fatalf("Failed to listen on insecure port: %v", err)
+	}
+	grpcServer := grpc.NewServer() // No TLS credentials
+	extproc.RegisterExternalProcessorServer(grpcServer, service)
+	reflection.Register(grpcServer)
+	if err := grpcServer.Serve(lis); err != nil {
+		log.Fatalf("Failed to serve gRPC on insecure port: %v", err)
+	}
+}
+
+type RequestHeadersHandler func(*extproc.HttpHeaders) (*extproc.ProcessingResponse, error)
+type ResponseHeadersHandler func(*extproc.HttpHeaders) (*extproc.ProcessingResponse, error)
+type RequestBodyHandler func(*extproc.HttpBody) (*extproc.ProcessingResponse, error)
+type ResponseBodyHandler func(*extproc.HttpBody) (*extproc.ProcessingResponse, error)
+type RequestTrailersHandler func(*extproc.HttpTrailers) (*extproc.ProcessingResponse, error)
+type ResponseTrailersHandler func(*extproc.HttpTrailers) (*extproc.ProcessingResponse, error)
+
+type HandlerRegistry struct {
+	RequestHeadersHandler   RequestHeadersHandler
+	ResponseHeadersHandler  ResponseHeadersHandler
+	RequestBodyHandler      RequestBodyHandler
+	ResponseBodyHandler     ResponseBodyHandler
+	RequestTrailersHandler  RequestTrailersHandler
+	ResponseTrailersHandler ResponseTrailersHandler
+}
+
+type GRPCCalloutService struct {
+	extproc.UnimplementedExternalProcessorServer
+	Handlers HandlerRegistry
+}
+
+func (s *GRPCCalloutService) Process(stream extproc.ExternalProcessor_ProcessServer) error {
+	for {
+		req, err := stream.Recv()
+		if err != nil {
+			return err
+		}
+
+		var response *extproc.ProcessingResponse
+		switch {
+		case req.GetRequestHeaders() != nil:
+			if s.Handlers.RequestHeadersHandler != nil {
+				response, err = s.Handlers.RequestHeadersHandler(req.GetRequestHeaders())
+			}
+		case req.GetResponseHeaders() != nil:
+			if s.Handlers.ResponseHeadersHandler != nil {
+				response, err = s.Handlers.ResponseHeadersHandler(req.GetResponseHeaders())
+			}
+		case req.GetRequestBody() != nil:
+			if s.Handlers.RequestBodyHandler != nil {
+				response, err = s.Handlers.RequestBodyHandler(req.GetRequestBody())
+			}
+		case req.GetResponseBody() != nil:
+			if s.Handlers.ResponseBodyHandler != nil {
+				response, err = s.Handlers.ResponseBodyHandler(req.GetResponseBody())
+			}
+		case req.GetRequestTrailers() != nil:
+			if s.Handlers.RequestTrailersHandler != nil {
+				response, err = s.Handlers.RequestTrailersHandler(req.GetRequestTrailers())
+			}
+		case req.GetResponseTrailers() != nil:
+			if s.Handlers.ResponseTrailersHandler != nil {
+				response, err = s.Handlers.ResponseTrailersHandler(req.GetResponseTrailers())
+			}
+		}
+
+		if err != nil {
+			return err
+		}
+
+		if response != nil {
+			if err := stream.Send(response); err != nil {
+				return err
+			}
+		}
+	}
+}
+
+func (s *GRPCCalloutService) HandleRequestHeaders(headers *extproc.HttpHeaders) (*extproc.ProcessingResponse, error) {
+	return &extproc.ProcessingResponse{
```

Comment:
This file doesn't have a line break after the function declarations, but the previous files did. Could you make it consistent?